providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -290,6 +290,7 @@ export class EthereumProvider implements IEthereumProvider {
       const session = await new Promise<SessionTypes.Struct | undefined>(
         async (resolve, reject) => {
           if (this.rpc.showQrModal) {
+            await this.loadWalletConnectModal();
             this.modal?.subscribeModal((state: { open: boolean }) => {
```

Comment:
Maybe refactor `loadWalletConnectModal` to be a "get instance" method and return the modal instance? Then you don't need to manually load the modal when you need it, because the only way to obtain it is to load it.